### ISSUE TITLE
pseudo_gravity_packing: Not adding all possible spheres and input radius not correctly used

### ISSUE THREAD

**Issue by heinsimon** [PMEAL/porespy#935]
# Problem
- pseudo_gravity_packing does not add all possible spheres to a simple image, that should result in a simple stacking of spheres
- pseudo_gravity_packing to get a sphere with radius R one needs to input R+1

# Example
```python

import porespy as ps
import numpy as np
import matplotlib.pyplot as plt

im_box=np.full((83,30,30),True)

# Bug in pseudo_gravity_packing: To get spheres with radius R one needs to input radius R+1
packing=ps.generators.pseudo_gravity_packing(im_box,r=15,clearance=0,maxiter=3,seed=42)

# Bug in pseudo_gravity_packing: There is room for 3 spheres, but only one is added.
# Correct spheres would be the one below
df={}
df['X']=np.array([23,51,79])
df['Y']=np.array([24,23,26])
df['Z']=np.array([25,26,23])
df['R']=np.array([14,14,14])


packing_sfc=ps.generators.spheres_from_coords(df,smooth=True,mode='contained')

print(f'np.sum pgp: {np.sum(packing)}')
print(f'np.sum sfc: {np.sum(packing_sfc)}')

fig,ax=plt.subplots(1,2, figsize=[12, 6])
ax[0].imshow(np.sum(packing,axis=1))
ax[0].set_title('packing bnd')
ax[1].imshow(np.sum(packing_sfc,axis=1))
ax[1].set_title('packing spheres from coords')
```

**Comment by jgostick** [comment]
I have fixed a lot of bugs in my recent PR (#922), so caught one of the ones you mentioned: the new version will add +1 to the radius if `smooth=True` and leave it as is if `smooth=False`. However, the code you have above still still shows a bug, but I can explain why.  If you reduce the `r` to 14, then you get 2 spheres. The reason is that to create the 'contained' edge condition I put a layer of `False` around the outside, but this technically makes the box 28 wide instead of 30, so spheres with `r=15` do not fit.  And the 3rd sphere is missing because the image is too short, only 83 instead of 90.  So if you do `imbox=np.ones([30, 30, 90])` and `r=14` it all fits, of do `imbox=np.ones([31, 31, 91])` with `r=15`.

Anyway, your bug report is valid, and I think I can fix it by padding the input `im` with `False` instead of inserting `False` along the edges of the existing image.  This is probably doable, so I will add it to the PR.

**Comment by jgostick** [comment]
I was mistaken, the code works fine. I do 'pad' the image, not overwrite the boundaries.  The reason that the spheres don't fit is because a sphere with a radius of 15 voxels is actually 31 voxels wide.  The following circle has a radius of 1:

```
[[0 1, 0],
 [1, 1, 1],
 [0, 1, 0]]
```
So with a radius of 1, the sphere is actually 3 voxels wide.  The 'radius extends from the center of the middle pixel, to the center of the outer pixel, but the digitization requires us to draw an entire pixel.
